lib/govspeak.rb

```diff
@@ -305,7 +305,7 @@ def render_image(image)
       <<~BODY
 
         <div class="address"><div class="adr org fn"><p markdown="1">
-        #{body.lstrip.gsub(/\\*\r?\n/, '<br />')}
+        #{body.lstrip.sub(/[\s\\]*\z/, '').gsub(/[ \\]*\r?\n/, '<br />')}
```

Comment:
**question:** regex syntax

Any reason for using `\s` in the `sub` but ` ` in the `gsub`?

Reply:
I wasn't sure whether this would be useful to try to document. And if so, where to do that. I could just add something to the commit message?

The significant difference is the use of the `\z`. It targets the very end of the `body` string, matching a point after any final line break characters.

Obviously in the `gsub` we're matching the ends of all lines by explicitly looking for their line break characters. But according to the test suite, the last line of an address block's content won't necessarily have a trailing line break.

So, the `gsub` matches space characters + slashes preceding a line break, while the `sub` matches any space characters + slashes + line break characters + any other whitespace characters preceding the very end of the `body`.

Reply:
I was mostly curious about the use of a space vs `\s` than the other differences, which I think are easier to work out

Reply:
Ah right. I know I could've been more concise but I thought I was answering your question,

> Any reason for using `\s` in the `sub` but ` ` in the `gsub`?

Do you mean that this would've been a reasonable answer: "Because `\s` matches line break characters as well as space characters."?

Reply:
Ah, didn't know that!